test/rules/hash_and_equals_test.dart

```diff
@@ -30,4 +33,15 @@ enum A {
       // no lint
     ]);
   }
+
+  test_extensionType_missingHash() async {
+    await assertDiagnostics(r'''
+extension type E(Object o) {
+  bool operator ==(Object other) => false;
+}
+''', [
+      // No lint.
```

Comment:
I'm similarly unconvinced that this test is particularly useful, though I guess I don't object too much here because there's nothing else to test.

Reply:
The goal here is to ensure we don't report a lint where there's already an error (since you can't redeclare `hash` or `==` in extension types).